pkgs/development/haskell-modules/configuration-ghc-8.10.x.nix

```
@@ -139,4 +139,10 @@ self: super: {
   # Not possible to build in the main GHC 9.0 package set
   # https://github.com/awakesecurity/spectacle/issues/49
   spectacle = doDistribute (markUnbroken super.spectacle);
+
+  # fails on missing doccheck dependency on ghc < 9.0
+  autodocodec = dontCheck super.autodocodec;
```

This should do the trick, the issue is just that the test suite is marked not buildable for GHC >= 9.0, so cabal2nix can't generate the dependencies for it.

```suggestion
  autodocodec = addBuildDepends [ self.doctest ] super.autodocodec;
```

---

After applying that, the test suite now fails due to a failing unit test. I'd like to just disable the tests for now as my package of interest `cabal2json` just works fine.

@NorfairKing , any idea why we are running into this here?
```
autodocodec> src/Autodocodec/Codec.hs:1140: failure in expression `JSON.parseMaybe scientificCodec (Number 3)'
autodocodec> expected: Just 3
autodocodec>  but got:
autodocodec>           ^
autodocodec>           <interactive>:621:17: error:
autodocodec>               • Couldn't match type ‘Codec Value Scientific Scientific’
autodocodec>                                with ‘Value -> JSON.Parser b’
autodocodec>                 Expected type: Value -> JSON.Parser b
autodocodec>                   Actual type: JSONCodec Scientific
autodocodec>               • In the first argument of ‘JSON.parseMaybe’, namely
autodocodec>                   ‘scientificCodec’
autodocodec>                 In the expression: JSON.parseMaybe scientificCodec (Number 3)
autodocodec>                 In an equation for ‘it’:
autodocodec>                     it = JSON.parseMaybe scientificCodec (Number 3)
autodocodec>               • Relevant bindings include
autodocodec>                   it :: Maybe b (bound at <interactive>:621:1)
autodocodec> src/Autodocodec/Codec.hs:1174: failure in expression `JSON.parseMaybe (parseJSONVia c) (Number 3)'
autodocodec> expected: Just 3
autodocodec>  but got: Just 3.0
autodocodec>                 ^
autodocodec> Examples: 112  Tried: 111  Errors: 0  Failures: 2
```

---

@DavHau doctest is broken on ghc > 9: https://github.com/sol/doctest/issues/327

---

This is about GHC 8.10.7 though?!

---

But this is built with ghc-8.10.x, and still it fails.

---

Oh wait yeah different error. I think this may just be a broken version. Best to disable tests for now. It builds on latest master I think...

---

Thanks, I updated the comment on the override. It should be good to go.

---

This exact autodocodec version (0.1.0.2) builds and tests fine using `stack test`. The error is likely because of some dependency mismatch introduced by nixpkgs. The application cabal2json which I'm actually interested in, works fine.

I do not have the power to release a new autodocodec. Would it be acceptable to mark autodocodec as broken, and make an override just for cabal2json that unbreaks it?

---

Hold on, I can release a newer version of autodocodec, one sec.

---

I just released 0.1.0.3 which shouldn't have this issue.

---

It'll be updated automatically by the next haskell-updates rotation, you don't need to do anything. I think someone will start one in the next few days, we're just all a little busy atm.

---

Should be available after rebasing on `haskell-updates` again (#181822).

---

Thanks, I rebased it and now the autodocodec package builds fine and tests pass.